backend/services/implementations/shiftSignupService.ts

```diff
@@ -0,0 +1,87 @@
+import { PrismaClient, Signup, SignupStatus } from "@prisma/client";
+
+import IShiftSignupService from "../interfaces/shiftSignupService";
+import {
+  CreateShiftSignupDTO,
+  ShiftSignupDTO,
+  ShiftSignupResponseDTO,
+  UpdateShiftSignupRequestDTO,
+} from "../../types";
+import logger from "../../utilities/logger";
+
+const prisma = new PrismaClient();
+
+const Logger = logger(__filename);
+
+class ShiftSignupService implements IShiftSignupService {
+  convertSignupToDTO = (signup: Signup): ShiftSignupDTO => {
+    return {
+      ...signup,
+      shiftId: String(signup.shiftId),
+      userId: String(signup.userId),
+    };
+  };
+
+  async getShiftSignupsForUser(
+    userId: string,
+  ): Promise<ShiftSignupResponseDTO[]> {
+    try {
+      const shiftSignups = await prisma.signup.findMany({
+        where: { userId: Number(userId) },
+      });
+      return shiftSignups.map((signup) => this.convertSignupToDTO(signup));
+    } catch (error) {
+      Logger.error(`Failed to shift signups. Reason = ${error.message}`);
+      throw error;
+    }
```

Comment:
For type-safe error handling, please use `getErrorMessage` from utilities/errorUtils.ts

```suggestion
    } catch (error: unknown) {
      Logger.error(`Failed to shift signups. Reason = ${getErrorMessage(error)}`);
      throw error;
    }
```